File: backend/bookingmovieticketapp/app/Models/Rating.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Rating extends Model
{
    use HasFactory;
    protected $table = 'ratings';
    protected $primaryKey = 'id';
    public $incrementing = true;
    public $timestamps = false;
    protected $fillable = [
        'movieid',
        'userid',
        'rating',
        'comment',
        'createdat'
    ];

    public function movie()
    {
        return $this->belongsTo(Movie::class, 'movieid');
    }

    public function user()
    {
        return $this->belongsTo(User::class, 'userid');
    }
}

File: frontendapp/lib/models/rating.dart
class Rating {
  final int? id;
  final int movieId;
  final int userId;
  final int rating;
  final String comment;
  final String createdAt;

  Rating({
    this.id,
    required this.movieId,
    required this.userId,
    required this.rating,
    required this.comment,
    required this.createdAt,
  });

  factory Rating.fromMap(Map<String, dynamic> map) => Rating(
    id: map['ID'],
    movieId: map['MOVIEID'],
    userId: map['USERID'],
    rating: map['RATING'],
    comment: map['COMMENT'],
    createdAt: map['CREATEDAT'],
  );

  Map<String, dynamic> toMap() => {
    'ID': id,
    'MOVIEID': movieId,
    'USERID': userId,
    'RATING': rating,
    'COMMENT': comment,
    'CREATEDAT': createdAt,
  };
}
